src/luarocks/deps.lua

```diff
@@ -400,6 +400,28 @@ local function rock_status(name, deps_mode)
    return installed and installed.." "..installation_type or "not installed"
 end
 
+--- Check depenendencies of a package and report any missing ones.
+-- @param name string: package name.
+-- @param version string: package version.
+-- @param dependencies table: array of dependencies.
+-- @param deps_mode string: Which trees to check dependencies for:
+-- "one" for the current default tree, "all" for all trees,
+-- "order" for all trees with priority >= the current default, "none" for no trees.
+function deps.report_missing_dependencies(name, version, dependencies, deps_mode)
+   local first_missing_dep = true
+
+   for _, dep in ipairs(dependencies) do
+      if not match_dep(dep, nil, deps_mode) then
+         if first_missing_dep then
+            util.printout(("Missing dependencies for %s %s:"):format(name, version))
+            first_missing_dep = false
+         end
+
+         util.printout(("   %s (%s)"):format(deps.show_dep(dep), rock_status(dep.name, deps_mode)))
```

Comment:
I just realized there's no point in showing the rock status, since they will all say `(not installed)`